dev-python/m2crypto/m2crypto-0.31.0-r2.ebuild

```diff
@@ -43,12 +43,18 @@ python_compile() {
 	# setup.py looks at platform.machine() to determine swig options.
 	# For exotic ABIs, we need to give swig a hint.
 	# https://bugs.gentoo.org/617946
-	# TODO: Fix cross-compiles
 	local -x SWIG_FEATURES=
 	case ${ABI} in
 		x32) SWIG_FEATURES="-D__ILP32__" ;;
 	esac
-	distutils-r1_python_compile --openssl="${EPREFIX}"/usr
+	distutils-r1_python_compile --openssl="${ESYSROOT}"/usr
+}
+
+src_prepare() {
+	# Switch out the compiler and it's arguments to the current build compiler instead of just the `cpp`
+	local CPP=`tc-getCPP | sed "s: :', ':"`
+	sed -i -e "s:^        pid = subprocess.Popen(\['cpp', '-Wp,-v', '-'\],:        pid = subprocess.Popen(\['${CPP}', '-Wp,-v', '-'\],:" setup.py
```

Comment:
@floppym: Unfortunately I was not able to simplify this more. The `tc-getCPP` command is exactly for this purpose -- to get the compiler even in a cross compiler environment.

Reply:
Sorry, but there must be a better solution. This is not getting merged.

Reply:
I'm open to any suggestion, but I don't understand what do you find questionable here exactly.You don't want `sed`, or you don't want to hardcode the compiler to `setup.py`? **Please be more constructive.**

This line has to be replaced either way, whatever we do. This is not a "hacky" solution, this is the correct compiler here. Do you want me to use [this](https://stackoverflow.com/questions/724664/python-distutils-how-to-get-a-compiler-that-is-going-to-be-used) method?

Reply:
No sed, and a patch that could be merged upstream please.

Reply:
Seems like someone has already fixed this issue.

https://github.com/mcepl/M2Crypto/commit/471582f654d65512a25839bb7f67eb92d0281f3d

https://github.com/mcepl/M2Crypto/commit/8a0a3e3cb37c83e9e4362dda60779fbe5f83fb5a

Reply:
You are right, I will make a patch at the end of the day, so that crossdev for 0.31.0-r2 also works, not only for the latest version.